src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2911,8 +2942,74 @@ public void Free()
             }
         }
 
+        internal ImmutableArray<SynthesizedSimpleProgramEntryPointSymbol> GetSimpleProgramEntryPoints()
+        {
+            if (this.ContainingSymbol is not NamespaceSymbol { IsGlobalNamespace: true }
+                || this.Name != WellKnownMemberNames.TopLevelStatementsEntryPointTypeName)
+            {
+                return ImmutableArray<SynthesizedSimpleProgramEntryPointSymbol>.Empty;
+            }
+
+            SimpleProgramEntryPointInfo? simpleProgramEntryPointInfo = _lazySimpleProgramEntryPoint;
+            if (simpleProgramEntryPointInfo != SimpleProgramEntryPointInfo.UninitializedSentinel)
+            {
+                return simpleProgramEntryPointInfo?.SimpleProgramEntryPoints ?? ImmutableArray<SynthesizedSimpleProgramEntryPointSymbol>.Empty;
```

Comment:
>return [](http://example.com/codeflow?start=16&length=6)

Consider changing the `if` to check `==` and moving the code below into the `if` block, so there is a single `return` since the return value is not trivial.